analyzer/include/ikos/analyzer/analysis/variable.hpp

```diff
@@ -407,6 +407,9 @@ class NamedShadowVariable final : public Variable {
   /// \brief Default constructor
   NamedShadowVariable(ar::Type* type, std::string name);
 
+  /// \brief Default constructor
+  NamedShadowVariable(ar::Type* type, llvm::StringRef name);
```

Comment:
Since `NamedShadowVariable` needs to store a `std::string` internally, it is better to let the caller do the conversion. Please remove this.